realm/realm-jni/src/io_realm_internal_TableQuery.cpp

```diff
@@ -1461,7 +1461,7 @@ JNIEXPORT void JNICALL Java_io_realm_internal_TableQuery_nativeIsNull(
         } else {
             switch (col_type) {
                 case type_Link:
-                    pQuery->and_query(src_table_ref->column<Link>(S(column_idx)).is_null());
+                    ThrowException(env, IllegalArgument, "isNull() by nested query for link field is not supported.");
```

Comment:
I wouldn't bother, because we couldn't be sure what C++ not_implemented should map to in Java, it could be a number of things. Just keep it as it is now then.